src/test/java/picard/analysis/TheoreticalSensitivityTest.java

```diff
@@ -216,13 +216,117 @@ public void testHetSensTargeted(final double expected, final File metricsFile) t
+    public Object[][] arbFracSensDataProvider() {
+        final File wgsMetricsFile = new File(TEST_DIR, "test_Solexa-332667.wgs_metrics");
+
+        // This test acts primarily as an intergration test.  The sample size of 100
+        // is not quite large enough to converge properly
+        return new Object[][] {
+                {0.90, wgsMetricsFile,       .5, 100},
+                {0.77, wgsMetricsFile,       .3, 100},
+                {0.29, wgsMetricsFile,       .1, 100},
+                {0.08, wgsMetricsFile,       .05, 100},
+        };
+    }
+
+    @Test(dataProvider = "TheoreticalSensitivityDataProvider")
+    public void testSensitivity(final double expected, final File metricsFile, final double alleleFraction, final int sampleSize) throws Exception {
+        // This tests Theoretical Sensitivity using distributions on both base quality scores
+        // and the depth histogram.
+
+        // We use a pretty forgiving tolerance here because for these tests
+        // we are not using large enough sample sizes to converge.
+        final double tolerance = 0.02;
+
+        final MetricsFile Metrics = new MetricsFile();
+        Metrics.read(new FileReader(metricsFile));
+        final List<Histogram> histograms = Metrics.getAllHistograms();
+        final Histogram depthHistogram = histograms.get(0);
+        final Histogram qualityHistogram = histograms.get(1);
+
+        final double[] qualityDistribution = TheoreticalSensitivity.normalizeHistogram(qualityHistogram);
+        final double[] depthDistribution = TheoreticalSensitivity.normalizeHistogram(depthHistogram);
+
+        final double result = TheoreticalSensitivity.theoreticalSensitivity(depthDistribution, qualityDistribution, sampleSize, 3, alleleFraction);
+
+        Assert.assertEquals(result, expected, tolerance);
+    }
+
+    @DataProvider(name = "equivalanceHetVsArbitrary")
+    public Object[][] equivalenceHetVsFull() {
+        final File wgsMetricsFile = new File(TEST_DIR, "test_Solexa-332667.wgs_metrics");
+        final File targetedMetricsFile = new File(TEST_DIR, "test_25103070136.targeted_pcr_metrics");
+
+        return new Object[][] {
+                // The sample sizes chosen here for these tests are smaller than what would normally be used
+                // in order to keep the test time low.  It should be noted that for larger sample sizes
+                // the values converge.
+                {wgsMetricsFile, 0.01, 300},
+                {targetedMetricsFile, 0.01, 50}
+        };
+    }
+
+    @Test (dataProvider = "equivalanceHetVsArbitrary")
+    public void testHetVsArbitrary(final File metricsFile, final double tolerance, final int sampleSize) throws Exception {
+        // This test compares Theoretical Sensitivity for arbitrary allele fractions with the theoretical het sensitivity
+        // model.  Since allele fraction of 0.5 is equivalent to a het, these should provide the same answer.
```

Comment:
Mention that there are minor differences in the two models, but they shouldn't matter here.  They might matter for low allele fraction, but not for het variants.